background.js

```diff
@@ -1,49 +1,53 @@
-browser.pageAction.onClicked.addListener(async () => {
-  await browser.tabs.executeScript({
-    code: `document.getElementsByClassName("download")[0].click();`,
-  });
-
-  await new Promise(r => setTimeout(r, 500));
+const clickElement = async querySelector => {
+  await browser.tabs.executeScript({code: `
+    if (document.querySelector("${querySelector}"))
+      document.querySelector("${querySelector}").click();
+  `});
+}
+
+const waitForElement = async querySelector => {
+  let [isElement] = await browser.tabs.executeScript({code: `document.querySelector("${querySelector}") !== null;`});
+  let timeLeft = 1000;
+
+  while (!isElement && timeLeft > 0) {
+    await new Promise(r => setTimeout(r, 100));
+    [isElement] = await browser.tabs.executeScript({code: `document.querySelector("${querySelector}") !== null;`});
+    timeLeft -= 100;
```

Comment:
Minor:

You could extract the magic `100` to a single variable to ensure that the timeout delay and the value that is subtracted from `timeLeft` are always the same. In larger programs, this would also have an additional benefit that it shows that this "100" is the same as the "100" below, meaning you can click "Find all references" for a variable and see where this variable matters. Moreover, it shows that all of these places should change at the same time when the variable changes - when using a variable you only need to update one place and there is no risk that you update the timeout but not the `timeLeft`

```ts
const retryDelay = 100;

await new Promise(r => setTimeout(r, retryDelay));
// ...
timeLeft -= retryDelay;
```